pkg/workceptor/command_test.go

```diff
@@ -0,0 +1,460 @@
+package workceptor_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ansible/receptor/pkg/workceptor"
+	"github.com/ansible/receptor/pkg/workceptor/mock_workceptor"
+	"github.com/golang/mock/gomock"
+)
+
+func statusExpectCalls(mockBaseWorkUnit *mock_workceptor.MockBaseWorkUnitForWorkUnit) {
+	statusLock := &sync.RWMutex{}
+	mockBaseWorkUnit.EXPECT().GetStatusLock().Return(statusLock).Times(2)
+	mockBaseWorkUnit.EXPECT().GetStatusWithoutExtraData().Return(&workceptor.StatusFileData{})
+	mockBaseWorkUnit.EXPECT().GetStatusCopy().Return(workceptor.StatusFileData{
+		ExtraData: &workceptor.CommandExtraData{},
+	})
+}
+
+func createTestSetup(t *testing.T) (workceptor.WorkUnit, *mock_workceptor.MockBaseWorkUnitForWorkUnit, *mock_workceptor.MockNetceptorForWorkceptor, *gomock.Controller, *workceptor.Workceptor) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+
+	mockBaseWorkUnit := mock_workceptor.NewMockBaseWorkUnitForWorkUnit(ctrl)
+	mockNetceptor := mock_workceptor.NewMockNetceptorForWorkceptor(ctrl)
+	mockNetceptor.EXPECT().NodeID().Return("NodeID")
+
+	w, err := workceptor.New(ctx, mockNetceptor, "/tmp")
+	if err != nil {
+		t.Errorf("Error while creating Workceptor: %v", err)
+	}
+
+	cwc := &workceptor.CommandWorkerCfg{}
+	mockBaseWorkUnit.EXPECT().Init(w, "", "", workceptor.FileSystem{}, nil)
+	workUnit := cwc.NewWorker(mockBaseWorkUnit, w, "", "")
+
+	return workUnit, mockBaseWorkUnit, mockNetceptor, ctrl, w
+}
+
+func TestCommandSetFromParams(t *testing.T) {
+	wu, mockBaseWorkUnit, _, _, _ := createTestSetup(t) //nolint:dogsled
+
+	paramsTestCases := []struct {
+		name          string
+		params        map[string]string
+		expectedCalls func()
+		errorCatch    func(error, *testing.T)
+	}{
+		{
+			name:   "no params with no error",
+			params: map[string]string{"": ""},
+			expectedCalls: func() {
+				mockBaseWorkUnit.EXPECT().GetStatusCopy().Return(workceptor.StatusFileData{
+					ExtraData: &workceptor.CommandExtraData{},
+				})
+			},
+			errorCatch: func(err error, t *testing.T) {
+				if err != nil {
+					t.Error(err)
+				}
+			},
+		},
+		{
+			name:   "params with error",
+			params: map[string]string{"params": "param"},
+			expectedCalls: func() {
+			},
+			errorCatch: func(err error, t *testing.T) {
+				if err == nil {
+					t.Error(err)
+				}
+			},
+		},
+	}
+
+	for _, testCase := range paramsTestCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			testCase.expectedCalls()
+			err := wu.SetFromParams(testCase.params)
+			testCase.errorCatch(err, t)
+		})
+	}
+}
+
+func TestUnredactedStatus(t *testing.T) {
```

Comment:
Can I get a few more tests in the block please and run them in `t.Parallel` . I would like to make sure that the new status lock wrappers you created do in fact lock the status object for running in parallel